core/task.cc

```diff
@@ -7,21 +7,43 @@
 
 #include "module.h"
 #include "tc.h"
+#include "worker.h"
 
 // Capture the default core command line flag.
 DECLARE_int32(c);
 
 struct cdlist_head all_tasks = CDLIST_HEAD_INIT(all_tasks);
 
 struct task_result task_scheduled(struct task *t) {
-  return t->m->RunTask(t->arg);
+  struct task_result ret = t->m->RunTask(t->arg);
+
+  // Depth first goes through all pending modules and services
+  while (ctx.gates_pending()) {
+    struct gate_task task = ctx.pop_ogate_and_packets();
+    gate *ogate = task.gate;
+    struct pkt_batch *next_packets = &(task.batch);
+
+#if TRACK_GATES
```

Comment:
No real need. This gets dropped by #128, which adds documentation somewhere else (unless we care about the commit history)